Menu Component

tewashi opened the pull request.

superMasterSword commented on src/components/menu/Menu.tsx:
```diff
@@ -0,0 +1,114 @@
+    variant='default',
+    options=[],
+    buttonLabel="MENU",
+    buttonColor='black',
```
I would recommend removing the button color defaults. By default it if color isn't provided it will use the ARGO color scheme.

tewashi replied:
Removed the button altogether.

superMasterSword reviewed (changes requested):
Don't have time to review the full thing right now, but I suggest removing the default button colors.

tewashi commented:
Removed the button from within the menu component. Added an open prop but also had to add a close prop because the MUI Menu "onClose" prop requires a function not a boolean. Example usage where the menu opens when a button is clicked, and closes when anything other than the menu itself is clicked:

```
function ToggleMenu() {
	const options = ['menu option', 'menu option', 'menu option']
	const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
	const openMenu = Boolean(anchorEl);
	const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
	  setAnchorEl(event.currentTarget);
	};
	const handleClose = () => {
	  setAnchorEl(null);
	};
	
	return (
		<div>
	            <Button variant='primary' label='TEST BUTTON' onClick={handleClick}></Button>
		    <Menu variant='checkmenu' options={options} open={openMenu} close={handleClose}></Menu>	
		</div>
	);
}
```
